java/jtest/TestH5Dffm.java

```
@@ -0,0 +1,1157 @@
+    private static final int RANK            = 2;
+
+    long H5fid  = hdf5_h.H5I_INVALID_HID();
+    long H5dsid = hdf5_h.H5I_INVALID_HID();
+    long H5did  = hdf5_h.H5I_INVALID_HID();
+
+    private void deleteFile(String filename)
+    {
+        File file = new File(filename);
+        if (file.exists()) {
+            try {
+                file.delete();
+            }
+            catch (SecurityException e) {
+                // Ignore
+            }
+        }
+    }
+
+    @Before
+    public void createH5file()
+    {
+        System.out.print(testname.getMethodName());
+
+        try (Arena arena = Arena.ofConfined()) {
+            // Create file
+            MemorySegment fileNameSegment = stringToSegment(arena, H5_FILE);
+            H5fid = hdf5_h.H5Fcreate(fileNameSegment, hdf5_h.H5F_ACC_TRUNC(), hdf5_h.H5P_DEFAULT(),
+                                     hdf5_h.H5P_DEFAULT());
+            assertTrue("H5Fcreate failed", isValidId(H5fid));
+
+            // Create dataspace
+            long[] dims               = {DIM_X, DIM_Y};
+            MemorySegment dimsSegment = allocateLongArray(arena, RANK);
+            copyToSegment(dimsSegment, dims);
+
+            H5dsid = hdf5_h.H5Screate_simple(RANK, dimsSegment, MemorySegment.NULL);
+            assertTrue("H5Screate_simple failed", isValidId(H5dsid));
+
+            // Create dataset
+            MemorySegment dsetNameSegment = stringToSegment(arena, DATASET_NAME);
+            H5did = hdf5_h.H5Dcreate2(H5fid, dsetNameSegment, hdf5_h.H5T_NATIVE_INT_g(), H5dsid,
+                                      hdf5_h.H5P_DEFAULT(), hdf5_h.H5P_DEFAULT(), hdf5_h.H5P_DEFAULT());
+            assertTrue("H5Dcreate2 failed", isValidId(H5did));
+
+            int flushResult = hdf5_h.H5Fflush(H5fid, hdf5_h.H5F_SCOPE_LOCAL());
+            assertTrue("H5Fflush failed", isSuccess(flushResult));
+        }
+    }
+
+    @After
+    public void deleteH5file()
+    {
+        closeQuietly(H5did, hdf5_h::H5Dclose);
```

Use of `closeQuietly` here is going to suppress any errors that occur during cleanup, which will silently paper over a large class of errors.  The post-test cleanup should detect if an error occurs during cleanup and cause the test to fail (assuming that the test succeed up to that point - if it failed earlier, it should avoid overwriting the earlier failure message).

---

This applies to the post-test cleanup in other FFM test files as well.

---

Most of the tests just continue anyway so no real change. These java tests are just testing access mostly rather then correctness, making sure the API and basic functionality work.

---

Do you mean to say that correctness of the FFM API isn't a priority for the initial release? If so, we should be very clear about that so that users don't try to use it under the impression it's well-supported. I would say that making sure the library doesn't leak memory or otherwise tarnish its state during resource cleanup in a way that's going to break later operations falls under basic functionality.

---

Historically, the java tests check for accessibility and functional passing of arguments. Leaks and resources should not be an issue, once the tests are confirmed to work.  For instance the test:
    public void testH5Dclose()
checks the close function for H5D for proper return value. Similar for other close calls.